code/controllers/subsystem/persistence.dm

```diff
@@ -0,0 +1,145 @@
+		)
+	)
+
+///Loads data at the start of the round
+/datum/controller/subsystem/persistence/Initialize()
+	LoadSeasonalItems()
+	return ..()
+
+///Stores data at the end of the round
+/datum/controller/subsystem/persistence/proc/CollectData()
+	return
+
+///Loads seasons data, advances seasons and saves the data
+/datum/controller/subsystem/persistence/proc/LoadSeasonalItems()
+	var/json_file = file("data/seasonal_items.json")
+	if(!fexists(json_file))
+		initialize_seasonal_items_file()
+	var/list/seasons_file_info = json_decode(file2text(json_file))
+
+	for(var/season_class in seasons_durations)
+		seasons_file_info = update_season_data(season_class, seasons_file_info)
+
+	fdel(json_file)
+	WRITE_FILE(json_file, json_encode(seasons_file_info))
+
+///Handles checking whether a season should advance and advancing it, along with setting up the seasons' values or procs for the round
+/datum/controller/subsystem/persistence/proc/update_season_data(season_class, list/seasons_file_info)
+	//loads a new entry for a season if one is missing
+	if(!LAZYACCESS(seasons_file_info, season_class)) //handles adding missing entries
+		var/list/template_season_entry = list(
+			"[season_class]" = list(LAST_UPDATE = 0, CURRENT_SEASON = 0) //values will be set afterwards
+		)
+		seasons_file_info += template_season_entry
+
+	//checks whether the season should be advanced
+	var/last_season_update_time = text2num(seasons_file_info[season_class][LAST_UPDATE])
+	var/time_since_last_update = world.realtime - last_season_update_time
+	if(time_since_last_update >= seasons_durations[season_class])
+		seasons_file_info[season_class][LAST_UPDATE] = world.realtime
+		seasons_file_info[season_class][CURRENT_SEASON]++
+
+	//Initializes the season datum that is chosen based on the current season
+	season_progress[season_class] = seasons_file_info[season_class][CURRENT_SEASON]
+	var/seasons_buckets_list_index = season_progress[season_class] % length(seasons_buckets[season_class]) + 1
+	var/season_typepath = seasons_buckets[season_class][seasons_buckets_list_index]
+	var/datum/season_datum/season_instance = new season_typepath
+
+	//Does stuff with the initialized season datum
+	season_items[season_class] = season_instance.item_list
+
+	//returns the updated season file data to write over the stored season file information
+	return seasons_file_info
+
+///Initializes the seasonal items file if it is missing
+/datum/controller/subsystem/persistence/proc/initialize_seasonal_items_file()
+	var/json_file = file("data/seasonal_items.json")
+	var/list/seasons_file_info = list()
+	WRITE_FILE(json_file, json_encode(seasons_file_info))
+
+///Used to make item buckets for the seasonal items system
+/datum/season_datum
+	///Name of the  season
+	var/name = "base season"
+	///Descrpition of the season
+	var/description = "The first season."
+	///Items that the season contains
+	var/list/item_list = list()
+
+/datum/season_datum/weapons/guns/rifle_var_one
```

Comment:
Ok change this var to seasonal so we can avoid using a reserved word and I'll merge 
Ditto for the rest below

Reply:
changed